### PR TITLE
[SuiteSparse] update GPU build, refresh 32-bit version

### PR DESCRIPTION
**Pull request by ViralBShah**

### CODE REVIEW

File: S/SuiteSparse/SuiteSparse/build_tarballs.jl [hunk 1]
```diff
@@ -1,5 +1,11 @@
 include("../common.jl")
 
+name = "SuiteSparse"
+
+sources = [sources,
```
Comment (giordano) [inline S/SuiteSparse/SuiteSparse/build_tarballs.jl:5]:
```suggestion
sources = [sources;
```

File: S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
 name = "SuiteSparse_GPU"
-version = v"5.8.1"
 
-# Collection of sources required to build SuiteSparse
-sources = [
-    GitSource("https://github.com/DrTimothyAldenDavis/SuiteSparse.git",
-              "1869379f464f0f8dac471edb4e6d010b2b0e639d"),
-    DirectorySource("./bundled"),
-]
+append!(sources, DirectorySource("./bundled"));
```
Comment (giordano) [inline S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl:5]:
```suggestion
sources = [
    sources;
    DirectorySource("./bundled"),
]
```

File: S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl [hunk 3]
```diff
@@ -87,12 +87,13 @@ append!(products, [
...
-append!(dependencies, [
-    Dependency("METIS_jll"),
-    Dependency("MPFR_jll"),
-    Dependency("GMP_jll"),
+dependencies = [
+    dependencies;
+    Dependency("METIS_jll")
+    Dependency("MPFR_jll")
+    Dependency("GMP_jll")
     BuildDependency(PackageSpec(name="CUDA_full_jll", version=v"10.0"))
```
Comment (ViralBShah) [inline S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl:95]: I suppose it does not like `v"10.0"`.

Comment (giordano) [reply S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl:95]: I think you need to specify one of the actual [registered versions](https://github.com/JuliaRegistries/General/blob/46718fe08a481d82ba7b5ed9b9a0154f8f9641ca/C/CUDA_full_jll/Versions.toml) (excluding the build number).  So for example `v"10.0.130"`

Comment (fingolfin) [reply S/SuiteSparse/SuiteSparse_GPU/build_tarballs.jl:95]: Indeed: this is not like the "compat" field in a Project.toml, it specifies an *exact* version to be used. If you want a range, you can assign a suitable `VersionSpec` instance here.

**Comment by ViralBShah** [comment]
@giordano If I want to test the binaries built here (before registering the packages), is there a way for me to do that?

**Comment by giordano** [comment]
Ugh, sorry, I missed it.  If you can build locally, the easiest thing is to build with
```
julia build_tarballs.jl --verbose --debug [triplet of your platform] --deploy=local
```
It'll generate the JLL package that you can `dev`

**Pull request merged**